libraries/Microsoft.Bot.Builder.Dialogs/Prompts/OAuthPrompt.cs

```diff
@@ -431,5 +442,10 @@ private bool ChannelSupportsOAuthCard(string channelId)
 
             return true;
         }
+
+        private bool RequiresLink(ITurnContext turnContext)
+        {
+            return !turnContext.Activity.ServiceUrl.Contains("http");
```

Comment:
bug alert 2: how should strng case affect this path? is service url guaranteed to be lowercase? Otherwise, how do you want to compare strings?

Reply:
urls tend to be case insensitive

Reply:
Good catch.